### PR TITLE
feat(user): Create directory, typeDefs, and CRUD resolvers

### LINKED ISSUE #27: feat(user): create user directory with typeDefs and resolvers
- create new users directory with exported typeDefs and resolvers files
- import typeDefs and resolvers into schema and include users typeDefs and resolvers in corresponding merge
- create type with required id and email string and password string and personId ID
- create queries for get all users and get one user by id
- create mutations for create, update, and delete one user
`
const usersTestData = [
    {
        id: "aff1e95b-1b6d-44df-babf-d0548e59cf27",
        email: 'test@fake-email.com',
        password: "$2a$10$T3NwukZ8rNDuXTiu.pqbveyx7EMBcipN1A30BcbagfTJeC5KItTmS",
        personID: "23048073-d8e0-4b40-bb6a-72c5a1198f0e"
    },
    {
        id: "37bede16-aa7d-43a2-a8a8-429db71a6019",
        email: 'another.test@fake-email.com',
        password: "$2a$10$/wRL0WgmMXW4XmRV35cxjuKykOFGRH4Nhy5XgyQcmaHb7h6dNc7DO",
        personID: "79c5a40e-011f-4d47-ab8b-2cb066acefe8"
    },
]
`

### PR DESCRIPTION
**Pull request by metamang**
closes #27

### CODE REVIEW

**Comment by metamang** [comment]
created a field resolver for person on User type as discussed in the issue conversation.

File: src/User/resolvers.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+
+const resolvers = {
+  Query: {
+    users: () => usersData,
+    user: (_parent, { id }) => usersData.find((user) => user.id === id),
+    userByEmail: (_parent, { email }) => usersData.find((user) => user.email === email)
+  },
+  User: {
+    person: (parent) => personsData.find((p) => p.id === parent.personId),
```
Comment (MetaMatt1) [inline src/User/resolvers.js:28]: great job on this field resolver.

File: src/User/resolvers.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      usersData.push(newUser);
+      return newUser;
+    },
+    updateUser: (_parent, { id, email, password, personId }) => {
+      const userIndex = usersData.findIndex((user) => user.id === id);
+      if (email) {
+        usersData[userIndex].email = email;
+      }
+      if (password) {
+        // TODO: Add encryption
```
Comment (MetaMatt1) [inline src/User/resolvers.js:46]: great job leaving a todo comment here for future work.

File: src/User/resolvers.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        usersData[userIndex].password = password;
+      }
+      if (personId) {
+        usersData[userIndex].personId = personId;
+      }
+      return usersData[userIndex]
+    },
+    deleteUser: (_parent, { id }) => {
+      usersData = usersData.filter((user) => user.id !== id);
+      return "User Destroyed"
```
Comment (MetaMatt1) [inline src/User/resolvers.js:56]: haha nice success message.

**Review by MetaMatt1 — APPROVED** [review]

**Review by RhyecaRiley — DISMISSED** [review]

**Pull request merged**